crates/superposition_proc_macros/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+use proc_macro::TokenStream;
```

Comment:
Why a proc_macro? Do we expect PlatformQuery like functionality elsewhere? Why not add PlatformQuery to superposition types and just implement the trait CustomQuery on it? The struct PlatformQuery can store it's pattern, capture group, etc in the struct itself

Reply:
The aim was to prevent repetition of logic related to  namespacing query params with some other prefix as we discussed in the meeting. 

CustomQuery does solve the repetition part but there are other boilerplate code like implementing FromRequest, etc, which I thought since we are trying to make this easily usable why not solve this completely.

Proc macros as you can see in code makes it way easier for use.

Reply:
Can we do it at a later point in time, when we need it?